### DISCUSSION TITLE
How can I track if the diff editor I opened is closed by the user or not

### DISCUSSION THREAD

**Discussion by kunal-singhh** [microsoft/vscode-discussions#1175] [Extension Development QnA] [answered]
So I am trying to give users an option to go through some file suggestions and either accept or deny them.  For this I am using the `vscode.diff` API but I want to be able to track n files and when the user closes on diff editor I open the next one.

I tried checking different editor events like `onDidCloseTextDocument`, `onDidChangeVisibleTextEditors` but could not find a way to check if 

1. The opened diff window that I opened is closed or not

```js

 vscode.commands.executeCommand(
    'vscode.diff',
    leftUri,
    rightUri,
    'Cached vs Disk Diff'
  );

```

Can someone please share if there is a way to listen to the close events for the diff editor specifically.

<img width="1175" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/112637427/e7a45a4a-b8bd-41d5-91d8-649907b38f9b">

**Comment by abrilurena** [comment]
Hi, @kunal-singhh! do you still need help with this?

**Reply by kunal-singhh** [reply]
Yes @abrilurena.

**Comment by tamuratak** [comment, marked as answer]
There is no straightforward way. See the following:

- https://github.com/microsoft/vscode/issues/15513
 
The following comments would be useful:

- https://github.com/microsoft/vscode/issues/15513#issuecomment-1245403215
- https://github.com/microsoft/vscode/issues/15513#issuecomment-1245585945
